pkg/beacon/relay/node.go

```diff
@@ -157,6 +158,61 @@ func (n *Node) ForwardSignatureShares(groupPublicKeyBytes []byte) {
 	n.netProvider.BroadcastChannelForwarderFor(name)
 }
 
+// ResumeSigningIfEligible enables a client to rejoin the ongoing signing process
+// after it was crashed or restarted and if it belongs to the signing group.
+func (n *Node) ResumeSigningIfEligible(
+	relayChain relayChain.Interface,
+	signing chain.Signing,
+) {
+	isEntryInProgress, err := relayChain.IsEntryInProgress()
+	if err != nil {
+		logger.Errorf(
+			"failed checking if an entry is in progress: [%v]",
+			err,
+		)
+		return
+	}
+
+	if isEntryInProgress {
+		previousEntry, err := relayChain.CurrentRequestPreviousEntry()
+		if err != nil {
+			logger.Errorf(
+				"failed to get a previous entry for current request: [%v]",
+				err,
+			)
+			return
+		}
+		entryStartBlock, err := relayChain.CurrentRequestStartBlock()
+		if err != nil {
+			logger.Errorf(
+				"failed to get a current relay request entry start block: [%v]",
```

Comment:
It should be `current relay request start block`; we either have relay request or relay entry.

We should also align those errors. For example:
- `failed to get a previous entry for the current request: [%v]"`
- `failed to get a start block for the current request: [%v]"`
- `failed to get group public key for the current request: [%v]`"

Reply:
done